File: 현세미/백준/용액.py
n = int(input())
arr = list(map(int, input().split()))

left = 0
right = n - 1
best_sum = float('inf')
best_pair = (0, 0)

while left < right:
    current_sum = arr[left] + arr[right]
    
    if abs(current_sum) < abs(best_sum):
        best_sum = current_sum
        best_pair = (arr[left], arr[right])
    
    if current_sum < 0:
        left += 1
    elif current_sum > 0:
        right -= 1
    else:
        break

print(best_pair[0], best_pair[1])

File: 김채린/추가문제/백준/용액.java
import java.util.*;
import java.io.*;

class Main{

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        long[] arr =Arrays.stream(br.readLine().split(" ")).mapToLong(Long::parseLong).toArray();
        Arrays.sort(arr);
        int left=0;
        int right=n-1;
        int resultLeft=0;
        int resultRight=0;
        long min = Long.MAX_VALUE;
        while (left<right){
            long a= arr[right] + arr[left];
            if (min>=Math.abs(a)){
                resultLeft=left;
                resultRight=right;
                min = Math.abs(a);
            }
            if (a>0){
                right--;
            }else if (a<0){
                left++;
            }else {
                System.out.println(arr[left] +" " +arr[right]);
                return;
            }
        }
        System.out.println(arr[resultLeft] +" " +arr[resultRight]);
    }
}
